### PR TITLE
add weakset polyfill

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/weak_set.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  add(val) {
+    this._weakmap.set(val, true);
+    return this;
+  }
+
+  delete(val) {
+    this._weakmap.delete(val);
+    return this;
```
Comment (mmun) [inline packages/ember-metal/lib/weak_set.js]: This should be `return this._weakmap.delete(val);`. It's a boolean that tells you whether an element was actually deleted or not.

**Review by mmun — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/weak_set.js (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/weak_set.js]: ok thanks

**Review by mmun — APPROVED** [review]

**Comment by bekzod** [comment]
needs restart I suppose

**Comment by mmun** [comment]
@bekzod I think it's a legit error for IE11:

```
not ok 10 IE 11.0 - Global error: Unable to get property 'set' of undefined or null reference at http://127.0.0.1:7774/3242/dist/ember.debug.js, line 23688
```

File: packages/ember-metal/lib/weak_set.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const HAS_WEAK_SET = typeof WeakSet === 'function';
+
+class WeakSetPolyFill {
+  construcor() {
```
Comment (mmun) [inline packages/ember-metal/lib/weak_set.js]: Typo. Should be `constructor` :)

Comment (bekzod) [reply packages/ember-metal/lib/weak_set.js]: oops late night typing

**Pull request merged**